mypy/checkmember.py

```diff
@@ -359,6 +359,8 @@ def type_object_type(info: TypeInfo, builtin_type: Callable[[str], Instance]) ->
 def type_object_type_from_function(init_or_new: FuncBase, info: TypeInfo,
                                    fallback: Instance) -> FunctionLike:
     signature = method_type_with_fallback(init_or_new, fallback)
+    signature = cast(FunctionLike,
+                     map_type_from_supertype(signature, info, init_or_new.info))
```

Comment:
Yeah, good point. Maybe some reasoning about why the cast is safe and a general description.